### PR TITLE
drivers: timer: telink_mtimer: Introduced RISCV mtimer for Telink SoC

### PR DESCRIPTION
**Pull request by mishadesh**
Basic RISCV mtimer doesn't support custom sleep mode implementation in SoC. 
Removed a sleep prevention blocker in RISCV mtimer modification for Telink SoC.

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
Do you have any clear explanation what is going on when we remove next lines
from MTIMER driver?
```
if (idle) {
	return;
}
```
As I see the only call with idle set to true is from
_zephyr/subsys/pm/pm.c_
```
z_set_timeout_expiry(ticks -
     k_us_to_ticks_ceil32(
	     z_cpus_pm_state[id].exit_latency_us),
		     true);
```	     
Here we have sleep time adjustment using exit-latency-us from .dts
So for example we have nothing to do during 1000mS. This time is bigger
than min-residency-us and we are going to enter suspend mode.
Exit from suspend takes some time due to HW (oscillator stabilization and
internal wake-up logic) additionally we should prepare/fix all HW peripheral drivers
before/after suspend/wake-up. Now we set this value to 1mS.
So we decrease sleep time by 1000ms - exit_latency_us.

But in RISC-V vendor I see that this time is not updated due to this return when idle.

I started with debugging using blinky sample with PM. And observed the next results:
1. without patch sys_clock_set_timeout input:
```
1001       0  // <- we have nothing to do for 1S - kernel marking it
 999       1  // <- idle process doing adjustment (which is actually skipped)
2147483647 0  // <- wake up we have just one task which is non-blocked so set next event to max
...this process repeats
```
2. with patch sys_clock_set_timeout input:
```
1001       0  // <- we have nothing to do for 1S - kernel marking it
 999       1  // <- idle process doing adjustment (now it's not skipped)
   1       0  // <- so early as result kernel is on idle without sleep due to min-residency-us
2147483647 0  // <- wake up we have just one task which is non-blocked so set next event to max
...this process repeats
```
So in my case it seems like we have consumption degradation.

To summarize:
- Do we really understand what we are doing? Why exit_latency_us is skipped in RISC-V vendors? There is some explanation in sys_clock_set_timeout but it's completely unclear for me. Additionnaly when we apply such changes we should check dalays latency additionnally to consumption this parameter is also important for timming sensetive applications (as example OT SSED)
- Maybe lets try to adjust min-residency-us as I remember it has been just set to random reasonably value.
- I observed CONFIG_RISCV_MACHINE_TIMER_MIN_DELAY which is 100 in our case. We have slow MTIMER clock so minimal sleep in our platform is ~3.1mS. Which is very poor let's try to set this value to 32 or lower.

**Comment by mishadesh** [comment]
https://github.com/zephyrproject-rtos/zephyr/issues/56980

I have explained almost the same here.
